### DISCUSSION TITLE
Domain decomposition method

### DISCUSSION THREAD

**Discussion by liangLL888** [idaholab/moose#21896] [Q&A General] [answered]
How to use non overlapping domain decomposition method to solve problems in MOOSE? Are there any examples?

**Comment by GiudGiud** [comment]
Hello

Most domain decomposition in MOOSE is non-overlapping. Overlapping is done as necessary (ghosting) when information is needed from neighbor domains. 

Any example or simulation in MOOSE may be run in parallel by running:
`mpirun -n <number of parallel processes> <application_executable> -i <input_file.i>`

I could not locate the mpi tutorial. You may find one online though

Guillaume

**Comment by lindsayad** [comment, marked as answer]
The default preconditioning type used by PETSc and MOOSE when you run in parallel is bjacobi which is not overlapping. If you wanted a domain decomposition preconditioning type with overlap then you would use -pc_type asm and then you can specify different amounts of overlap with ‘-pc_asm_overlap‘. The default value for that option is 1. A value of 0 would be equivalent to bjacobi (block Jacobi).
